stl/src/stacktrace.cpp

```diff
@@ -0,0 +1,199 @@
+                is_description_valid = false;
+                is_line_valid        = false;
+                last_address         = address;
+            }
+        }
+
+        void initialize_line(const void* const address) {
+            clear_if_wrong_address(address);
+
+            if (!is_line_valid) {
+                bool success =
+                    try_initialize()
+                    && SymGetLineFromAddr(process_handle, reinterpret_cast<uintptr_t>(address), &displacement, &line);
+                if (!success) {
+                    line.FileName   = nullptr;
+                    line.LineNumber = 0;
+                    displacement    = 0;
+                }
+                is_line_valid = true;
+            }
+        }
+
+        static constexpr SYMBOL_INFO init_symbol_info() {
+            SYMBOL_INFO result  = {};
+            result.SizeOfStruct = sizeof(SYMBOL_INFO);
+            result.MaxNameLen   = max_line_size;
+            return result;
+        }
+
+        SRWLOCK srw               = SRWLOCK_INIT;
+        HANDLE process_handle     = nullptr;
+        bool initialized          = false;
+        bool initialize_attempted = false;
+        const void* last_address  = nullptr;
+        bool is_description_valid = false;
+        bool is_line_valid        = false;
+        IMAGEHLP_LINE line        = {sizeof(IMAGEHLP_LINE)};
+        DWORD displacement        = 0;
+        SYMBOL_INFO info          = init_symbol_info();
+        wchar_t buffer[max_line_size];
+    };
+
+    static stacktrace_global_data_t stacktrace_global_data;
+
+} // namespace
+
+[[nodiscard]] unsigned short __stdcall __std_stacktrace_capture(
+    unsigned long _FramesToSkip, unsigned long _FramesToCapture, void** _BackTrace, unsigned long* _BackTraceHash) {
+#ifdef _DEBUG
+    _FramesToSkip += 1; // compensate absense of tail call optimization here
+#endif
+    return CaptureStackBackTrace(_FramesToSkip, _FramesToCapture, _BackTrace, _BackTraceHash);
+}
+
+[[nodiscard]] std::string __stdcall __std_stacktrace_description(const void* _Address) {
+    return stacktrace_global_data.description(_Address);
+}
+
+[[nodiscard]] std::string __stdcall __std_stacktrace_source_file(const void* _Address) {
+    return stacktrace_global_data.source_file(_Address);
+}
+
+[[nodiscard]] unsigned __stdcall __std_stacktrace_source_line(const void* _Address) {
+    return stacktrace_global_data.source_line(_Address);
+}
+
+[[nodiscard]] std::string __stdcall __std_stacktrace_address_to_string(const void* _Address) {
+    std::string result;
+    auto line = __std_stacktrace_source_line(_Address);
+    auto desc = __std_stacktrace_description(_Address);
+
+    if (line != 0) {
+        result = std::format("{}({})", __std_stacktrace_source_file(_Address), line);
+    }
+
+    if (!desc.empty()) {
+        if (result.empty()) {
+            result = std::move(desc);
+        } else {
+            result = std::format("{}: {}", result, desc);
+        }
+    }

```

Comment:
Wouldn't something like this be clearer and faster?
```suggestion
    if (!desc.empty()) {
        if (line == 0) {
            result = std::move(desc);
        } else {
            result = std::format("{}({}): {}", __std_stacktrace_source_file(_Address), line, desc);
        }
    }
```
It would even be possible to remove `result` altogether:
```suggestion
    if (!desc.empty()) {
        if (line == 0) {
            return desc;
        } else {
            return std::format("{}({}): {}", __std_stacktrace_source_file(_Address), line, desc);
        }
    }
```